packages/framework/src/Component/Redis/RedisVersionsFacade.php

```diff
@@ -32,18 +32,21 @@ public function cleanOldCache(): void
     {
         $versionPattern = '[0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]*'; // redis pattern is a glob style and doesn't support repetitions
 
-        $all = $this->globalClient->keys($versionPattern);
-        $toRemove = [];
-
         $prefix = (string)$this->globalClient->getOption(Redis::OPT_PREFIX);
         $currentVersionPrefix = $prefix . $this->currentVersion;
-        foreach ($all as $key) {
-            if (strpos($key, $currentVersionPrefix) === false) {
-                $keyWithoutPrefix = substr($key, strlen($prefix)); // redis returns keys including prefix but needs them without prefix during removing
-                $toRemove[] = $keyWithoutPrefix;
+
+        $this->globalClient->setOption(Redis::OPT_SCAN, Redis::SCAN_RETRY);
+        $iterator = null;
+
+        while ($keys = $this->globalClient->scan($iterator, $versionPattern, 10000)) {
+            $toRemove = [];
+            foreach ($keys as $key) {
```

Comment:
hmm what about to use rather array_map then?

Reply:
Well, neither of these don't fit the need. I could somehow rewrite it with a combination of `array_filter` and `array_map`, but for me, it's hard to read then.
Or can you write an example of your idea?